pkg/destinations/bigquery/bigquery.go

```diff
@@ -0,0 +1,98 @@
+package bigquery
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/rs/zerolog/log"
+	"github.com/scratchdata/scratchdata/util"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+)
+
+type BigQueryServer struct {
+	DatasetID             string `mapstructure:"bigquery_dataset_id"`
+	CredentialsJsonString string `mapstructure:"bigquery_credentials_json"`
+
+	// this will be implemented during runtime
+	ProjectId *string
+
+	GCSBucketName  string `mapstructure:"gcs_bucket_name"`
+	GCSAccessKey   *string
+	GCSAccessKeyId *string
+	GCSFilePrefix  string `mapstructure:"gcs_file_prefix"`
+
+	DeleteFromGCS bool `mapstructure:"delete_from_gcs"`
+
+	Credentials *google.Credentials
+	conn        *bigquery.Client
+}
+
+func parseCredentials(credentialsString string) (map[string]interface{}, error) {
+	var credentials map[string]interface{}
+	err := json.Unmarshal([]byte(credentialsString), &credentials)
+	if err != nil {
+		return nil, err
+	}
+	return credentials, nil
+}
+
+func openConn(s *BigQueryServer) (*bigquery.Client, error) {
+	ctx := context.Background()
+	credentialsJson, err := parseCredentials(s.CredentialsJsonString)
```

Comment:
You don't need this. You can get the ProjectID from `google.CredentialsFromJSON()`

https://pkg.go.dev/golang.org/x/oauth2/google#Credentials